src/main/java/rex/task/TaskList.java

```diff
@@ -0,0 +1,250 @@
+package rex.task;
+
+import rex.exception.InvalidInputException;
+import rex.util.Parser;
+import rex.util.Storage;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/**
+ * The {@code TaskList} class manages a list of tasks, including adding, retrieving, updating,
+ * and deleting tasks. It also handles the loading and saving of tasks to persistent storage.
+ */
+public class TaskList {
+    private Storage storage;
+    private ArrayList<Task> list;

```

Comment:
Would it be better if list is in plural?

Reply:
Nice spot, will rename it to tasks instead for clarity!